app/controllers/katello/api/v2/host_subscriptions_controller.rb

```diff
@@ -35,6 +37,42 @@ def events
       respond_for_index :collection => collection
     end
 
+    api :POST, "/hosts/subscriptions/register", N_("Simulate a host subscription-manager registration")
```

Comment:
On Mon, Mar 21, 2016 at 5:17 PM, Justin Sherrill <notifications@github.com>
wrote:

> In app/controllers/katello/api/v2/host_subscriptions_controller.rb
> <https://github.com/Katello/katello/pull/5831#discussion_r56899165>:
>
> > @@ -35,6 +37,42 @@ def events
> >        respond_for_index :collection => collection
> >      end
> >
> > +    api :POST, "/hosts/subscriptions/register", N_("Simulate a host subscription-manager registration")
>
> @ehelms <https://github.com/ehelms> are you saying:
>
> a) there is no need for creating a host with subscription information out
> of using 'subscription-manager register'
>



> or
> b) if someone wants to create a host with subscription information they
> should use the rhsm api?
>
I still haven't seen a compelling reason that users shouldn't use that API
given that is the API we actually support. Further, right now the code
requires us to duplicate code across them.

What is the actual use case? When I hear an API for "testing" or to
"simulate" as the description of the API says, that does not ring true to
me an API we should be supporting.

Do users need to be able to create host records that they later attach a
subscription facet to? Do they need to be able to create host records with
a "staged" subscription facet similar to an unbuilt or unmanaged host? If
facet end points are going to support creating a host through them, then I
am less opposed to having an API to do so. If the goal is to have a unified
API to create a host, then this would go against that or would serve as a
stand-in until that happens.

or
> c) something else?
>
—
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/Katello/katello/pull/5831/files/98237cdb5a53517ec253055f3b2ba023ffb9ba81#r56899165>
>



-- 
Eric D. Helms
Red Hat Engineering
Ph.D. Student - North Carolina State University